api/storage_service.cc

```diff
@@ -722,18 +722,26 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
     ss::remove_node.set(r, [&ss](std::unique_ptr<request> req) {
         auto host_id = validate_host_id(req->get_query_param("host_id"));
         std::vector<sstring> ignore_nodes_strs= split(req->get_query_param("ignore_nodes"), ",");
-        auto ignore_nodes = std::list<gms::inet_address>();
+        auto ignore_nodes = std::list<locator::host_id_or_endpoint>();
         for (std::string n : ignore_nodes_strs) {
             try {
                 std::replace(n.begin(), n.end(), '\"', ' ');
                 std::replace(n.begin(), n.end(), '\'', ' ');
                 boost::trim_all(n);
                 if (!n.empty()) {
-                    auto node = gms::inet_address(n);
-                    ignore_nodes.push_back(node);
+                    auto hoep = locator::host_id_or_endpoint(n);
+                    if (!ignore_nodes.empty()) {
+                        if (hoep.has_host_id() && !ignore_nodes.front().has_host_id()) {
+                            throw std::runtime_error("Expected host UUID. Use the same method for all nodes to ignore: either Host IDs or ip addresses.");
+                        }
+                        if (hoep.has_endpoint() && !ignore_nodes.front().has_endpoint()) {
```

Comment:
This can be condensed into a single if:
```
if (hoep.has_host_id() != ignore.nodes.front().has_host_id()) {
```

Reply:
I wanted the error to be more user friendly about what is expected, but maybe it's not that important, after all the list should be symmetrical.

Reply:
done in v3